### PR TITLE
Update Spring Cloud Azure README file

### PR DESCRIPTION
**Pull request by saragluna**
# Description

This PR will update the Spring Cloud Azure readme file to:
- Include the supported versions of Spring Cloud Azure
- Add section for Spring Boot 3.x
- Add description for users who runs into the signed jar issue when building native executable. 

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by Netyyyy — APPROVED** [review]

File: sdk/spring/README.md [hunk 1]
```diff
@@ -147,6 +148,45 @@ If you’re a Maven user, add our BOM to your pom.xml `<dependencyManagement>` s
 ```
 [//]: # ({x-version-update-end})
 
+## Spring Boot 3 Support
+
+The source code of Spring Cloud Azure for Spring Boot 3.x can be found on the [feature/spring-boot-3](https://github.com/Azure/azure-sdk-for-java/tree/feature/spring-boot-3) branch.
+
+#### Disable the JAR signature verification for the native image generation
+
+You have to disable the JAR signature verification to be able to generate a native image ([see](https://github.com/Azure/azure-sdk-for-java/issues/30320)).
```
Comment (jeanbisutti) [inline sdk/spring/README.md:157]: Azure SDK JARs are signed. [Spring Boot 3 does not support today signed JARs](https://github.com/Azure/azure-sdk-for-java/issues/30320) when you run your application with [AOT mode on a JVM](https://docs.spring.io/spring-boot/docs/current/reference/html/deployment.html#deployment.efficient.aot) or you [build an native image](https://docs.spring.io/spring-boot/docs/current/reference/html/native-image.html).

File: sdk/spring/README.md [hunk 2]
```diff
@@ -147,6 +148,45 @@ If you’re a Maven user, add our BOM to your pom.xml `<dependencyManagement>` s
 ```
 [//]: # ({x-version-update-end})
 
+## Spring Boot 3 Support
+
+The source code of Spring Cloud Azure for Spring Boot 3.x can be found on the [feature/spring-boot-3](https://github.com/Azure/azure-sdk-for-java/tree/feature/spring-boot-3) branch.
+
+#### Disable the JAR signature verification for the native image generation
```
Comment (jeanbisutti) [inline sdk/spring/README.md:155]:
```suggestion
#### Spring AOT and Spring native images
```

File: sdk/spring/README.md [hunk 3]
```diff
@@ -147,6 +148,45 @@ If you’re a Maven user, add our BOM to your pom.xml `<dependencyManagement>` s
...
+
+You have to disable the JAR signature verification to be able to generate a native image ([see](https://github.com/Azure/azure-sdk-for-java/issues/30320)).
+
+You can do this in the following way for GraalVM Native Build Tools:
```
Comment (jeanbisutti) [inline sdk/spring/README.md:159]:
```suggestion
You can disable the JAR signature verification in the following way for GraalVM Native Build Tools:
```

File: sdk/spring/README.md (hunk 1 above)
Comment (jeanbisutti) [inline sdk/spring/README.md:157]:
```suggestion
Azure SDK JARs are signed. https://github.com/Azure/azure-sdk-for-java/issues/30320 when you run your application with [AOT mode on a JVM](https://docs.spring.io/spring-boot/docs/current/reference/html/deployment.html#deployment.efficient.aot) or you [build an native image](https://docs.spring.io/spring-boot/docs/current/reference/html/native-image.html).
```

File: sdk/spring/README.md [hunk 4]
```diff
@@ -147,6 +148,45 @@ If you’re a Maven user, add our BOM to your pom.xml `<dependencyManagement>` s
 ```
 [//]: # ({x-version-update-end})
 
+## Spring Boot 3 Support
+
+The source code of Spring Cloud Azure for Spring Boot 3.x can be found on the [feature/spring-boot-3](https://github.com/Azure/azure-sdk-for-java/tree/feature/spring-boot-3) branch.
+
+#### Spring AOT and Spring native images
+
+Azure SDK JARs are signed. https://github.com/Azure/azure-sdk-for-java/issues/30320 when you run your application with [AOT mode on a JVM](https://docs.spring.io/spring-boot/docs/current/reference/html/deployment.html#deployment.efficient.aot) or you [build an native image](https://docs.spring.io/spring-boot/docs/current/reference/html/native-image.html).
```
Comment (jeanbisutti) [inline sdk/spring/README.md:157]:
```suggestion
Azure SDK JARs are signed. [Spring Boot 3 does not support today signed JARs](https://github.com/Azure/azure-sdk-for-java/issues/30320) when you run your application with [AOT mode on a JVM](https://docs.spring.io/spring-boot/docs/current/reference/html/deployment.html#deployment.efficient.aot) or you [build an native image](https://docs.spring.io/spring-boot/docs/current/reference/html/native-image.html).
```

File: sdk/spring/README.md [hunk 5]
```diff
@@ -147,6 +148,45 @@ If you’re a Maven user, add our BOM to your pom.xml `<dependencyManagement>` s
 ```
 [//]: # ({x-version-update-end})
 
+## Spring Boot 3 Support
+
+The source code of Spring Cloud Azure for Spring Boot 3.x can be found on the [feature/spring-boot-3](https://github.com/Azure/azure-sdk-for-java/tree/feature/spring-boot-3) branch.
+
+#### Spring AOT and Spring native images
+
+Azure SDK JARs are signed. [Spring Boot 3 does not support today signed JARs](https://github.com/Azure/azure-sdk-for-java/issues/30320) when you run your application with [AOT mode on a JVM](https://docs.spring.io/spring-boot/docs/current/reference/html/deployment.html#deployment.efficient.aot) or you [build an native image](https://docs.spring.io/spring-boot/docs/current/reference/html/native-image.html).
```
Comment (jeanbisutti) [inline sdk/spring/README.md:157]:
```suggestion
Azure SDK JARs are signed. [Spring Boot 3 does not support today signed JARs](https://github.com/Azure/azure-sdk-for-java/issues/30320) when you run your application with [AOT mode on a JVM](https://docs.spring.io/spring-boot/docs/current/reference/html/deployment.html#deployment.efficient.aot) or you [build a native image](https://docs.spring.io/spring-boot/docs/current/reference/html/native-image.html).
```

**Review by jeanbisutti — APPROVED** [review]

**Pull request merged**